src/Middleware/SpaServices.Extensions/src/AngularCli/AngularCliMiddleware.cs

```diff
@@ -36,10 +37,25 @@ internal static class AngularCliMiddleware
                 throw new ArgumentException("Cannot be null or empty", nameof(npmScriptName));
             }
 
-            // Start Angular CLI and attach to middleware pipeline
-            var appBuilder = spaBuilder.ApplicationBuilder;
-            var logger = LoggerFinder.GetOrCreateLogger(appBuilder, LogCategoryName);
-            var angularCliServerInfoTask = StartAngularCliServerAsync(sourcePath, npmScriptName, logger);
+            // We have to start the server by ourself if there wasn't any port specified
+            // or the specified port is still free (unused).
+            var shouldStartServer = !spaPort.HasValue || TcpPortFinder.TestPortAvailability(spaPort.Value);
```

Comment:
It feels like in the case that a port was specified but it's available we should either

1. Start the server on that port (below starts the server on an arbitrary port).
2. Fail with a message informing you that your server didn't start/was stopped.

Reply:
My use case for this PR is, that I want to be able to either let dotnet start the npm script or start it by myself such that webpack doesn't have to recompile everything, when dotnet has to rebuild some files (and restart the dotnet server). Specifying the port is only relevant for the second case. If the script isn't running and dotnet starts it by itself, I don't care about the port being used.
However to be consistent, approach 1 is in my opinion the best way to go.